gateway/handle_http_auth.go

```diff
@@ -239,7 +239,10 @@ func sourceToRequestContext(s backendconfig.SourceT) *gwtypes.AuthRequestContext
 		ReplaySource:   s.IsReplaySource(),
 		Source:         s,
 	}
-	arctx.SourceDetails.SourceDefinition = s.SourceDefinition
+	arctx.SourceDetails.SourceDefinition.Name = s.SourceDefinition.Name
+	arctx.SourceDetails.SourceDefinition.ID = s.SourceDefinition.ID
+	arctx.SourceDetails.SourceDefinition.Category = s.SourceDefinition.Category
+	arctx.SourceDetails.SourceDefinition.Type = s.SourceDefinition.Type
```

Comment:
Why are we not copying source definition options? If we add some other field in the future in source definition, noone will remember to include this here. If it is absolutely necessary, then it is best to introduce a SourceDefinition method, which has higher chances to be seen and be updated.